core/src/main/java/brooklyn/location/access/PortForwardManager.java

```diff
@@ -46,6 +47,42 @@
 @Beta
 public interface PortForwardManager extends Location {
 
+    @Beta
+    class AssociationMetadata {
+        private final String publicIpId;
+        private final HostAndPort publicEndpoint;
+        private final Location location;
+        private final int privatePort;
+
+        public AssociationMetadata(String publicIpId, HostAndPort publicEndpoint, Location location, int privatePort) {
```

Comment:
Worth a javadoc to say: `Users are discouraged from calling this constructor; the signature may change in future releases. Instead, instances will be created automatically by Brooklyn to be passed to the {@link AssociationListener#onEvent(AssociationMetadata)} method.`